### PR TITLE
WIP: Ghc 8.4

### PR DESCRIPTION
**Pull request by phadej**
I hopefully did most of the boring parts.

Current state:

```
[117 of 121] Compiling Language.Haskell.Liquid.Model ( src/Language/Haskell/Liquid/Model.hs, /home/ogre/Documents/other-haskell/liquidhaskell/dist-newstyle/build/x86_64-linux/ghc-8.4.3/liquidhaskell-0.8.2.4/build/Language/Haskell/Liquid/Model.o )

src/Language/Haskell/Liquid/Model.hs:332:31: error:
    • Couldn't match type ‘(TyThing, Fixity, [ClsInst], [FamInst],
                            GHC.SDoc)’
                     with ‘(a0, b0, [ClsInst], d0)’
```

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -3,9 +3,12 @@
 
 packages: .
           ./liquid-fixpoint
+          ./text-format
```
Comment (phadej) [inline cabal.project:6]: https://github.com/bos/text-format/pull/21 local.

Let's see what Hackage Trustees can do with that.

**Comment by ranjitjhala** [comment]
Sweet -- thanks for starting this @phadej  -- have some cycles now so will try to wrap it up soon!

**Comment by phadej** [comment]
@ranjitjhala no worries, thank @nikivazou for mentioning this task in her ZuriHac-keynote :)

I'll return to `text-format` issue next week.

**Comment by ranjitjhala** [comment]
Ok, I got this to build properly etc. and addressed the TODO GHC-8.4 but it appears to hit this

```
rjhala@borscht ~/r/s/liquidhaskell (phadej-ghc-8.4)> stack exec -- liquid tests/pos/Abs.hs
LiquidHaskell Version 0.8.4.0, Git revision 3f8facc34af23e623603b26526e7074c88cdbb5a (dirty)
Copyright 2013-18 Regents of the University of California. All Rights Reserved.

liquid:
lookupSymbol failed in relocateSection (RELOC_GOT)
/Users/rjhala/.stack/programs/x86_64-osx/ghc-8.4.3/lib/ghc-8.4.3/integer-gmp-1.0.2.0/HSinteger-gmp-1.0.2.0.o: unknown symbol `___gmp_rands'
liquid: liquid: unable to load package `integer-gmp-1.0.2.0'
```

which AFAICT appears to be this GHC bug ...

https://ghc.haskell.org/trac/ghc/ticket/15105

So as directed above I simply *deleted* the file

```
/Users/rjhala/.stack/programs/x86_64-osx/ghc-8.4.3/lib/ghc-8.4.3/integer-gmp-1.0.2.0/HSinteger-gmp-1.0.2.0.o 
```

After this I am stuck on the below:

---

In doing so, I find that I can no longer resolve
(i.e. get the `TyThing`, and hence `TyCon`)
corresponding to the name:

   GHC.Prim.Int#

It seems like in older versions, we had

   λ> :i GHC.Prim.Int#
   data GHC.Prim.Int#     -- Defined in ‘GHC.Prim’
   λ> :k GHC.Prim.Int#
   GHC.Prim.Int# :: #

but in GHC 8.4 this is changed so:

   λ> :i GHC.Prim.Int#
   data GHC.Prim.Int# :: TYPE 'GHC.Types.IntRep
      -- Defined in ‘GHC.Prim’
   λ> :k GHC.Prim.Int#
   GHC.Prim.Int# :: TYPE 'GHC.Types.IntRep

Does the above indicate that in fact, `GHC.Prim.Int#`
DOES NOT (any longer) correspond to a `TyCon`? If so,
what does it correspond to? i.e. how is it represented
as a `Type` in GHC?
---

[This is in the branch `phadej-ghc-8.4`]

**Comment by ranjitjhala** [comment]
Closing this as its folded into #1353

**Pull request closed without merging**